tools/pipelines/templates/build-npm-package.yml

```diff
@@ -425,6 +435,87 @@ stages:
+                  echo "@fluid-internal:registry=${{ variables.devFeed }}" >> ./.npmrc
+                  echo "@ff-internal:registry=https://pkgs.dev.azure.com/fluidframework/internal/_packaging/build/npm/registry/" >> ./.npmrc
+                  echo "@microsoft:registry=https://office.pkgs.visualstudio.com/_packaging/Office/npm/registry/" >> ./.npmrc
+                  echo "always-auth=true" >> ./.npmrc
+                  cat .npmrc
+
+            - task: npmAuthenticate@0
+              displayName: 'npm authenticate (internal feed)'
+              inputs:
+                workingFile: ${{ variables.testWorkspace }}/.npmrc
+
+            - task: npmAuthenticate@0
+              displayName: 'npm authenticate (office feed)'
+              inputs:
+                workingFile: ${{ variables.testWorkspace }}/.npmrc
+                customEndpoint: 'Office Packages - Readonly'
+
+            - task: Bash@3
+              displayName: 'List report.json and .npmrc'
+              inputs:
+                targetType: 'inline'
+                workingDirectory:  ${{ parameters.buildDirectory }}
+                script: |
+                  echo "Build Directory is ${{ parameters.buildDirectory }}"
+                  ls -la '${{ parameters.buildDirectory }}/artifacts/bundleAnalysis/@fluid-example/bundle-size-tests';
+                  cat '${{ variables.testWorkspace }}/.npmrc';
+
+            - task: Bash@3
+              displayName: 'Prepare telemetry-generator for bundle size'
+              inputs:
+                targetType: 'inline'
+                workingDirectory:  ${{ variables.toolAbsolutePath }}
+                script: |
+                  ls -la '${{ variables.testWorkspace }}';
+                  ls -la '${{ parameters.buildDirectory }}';
+                  cp ${{ variables.testWorkspace }}/.npmrc . ;
+                  npm install @ff-internal/aria-logger;
+                  npm i;
+                  npm run build:compile;
+
+            - task: Bash@3
+              displayName: Write bundle sizes measurements to Aria/Kusto
+              condition: succeededOrFailed()
+              inputs:
+                targetType: 'inline'
+                workingDirectory: ${{ variables.toolAbsolutePath }}
+                script: |
+                  echo "Writing the following performance tests results to Aria/Kusto"
+                  echo "Report Size:"
+                  ls -la '../../examples/utils/bundle-size-tests/bundleAnalysis/report.json';
+                  echo "Tools Path:"
+                  ls -la '../../tools/telemetry-generator/dist/handlers';
+                  node --require @ff-internal/aria-logger bin/run --handlerModule $(Build.SourcesDirectory)/tools/telemetry-generator/dist/handlers/bundleSizeHandler.js --dir '../../artifacts/bundleAnalysis/@fluid-example/bundle-size-tests';
+                  echo "Removing .npmrc"
+                  rm ../../tools/telemetry-generator/.npmrc;
```

Comment:
yes it is necessary as it fails a step further if I do not add it :)

Reply:
Curious what fails afterwards. It shouldn't complain about extraneous files, I think, because we added `.npmrc` to the exclusions... unless maybe the exclusion only applies to the root `.npmrc`. In which case yeah, let's delete it here.

Reply:
Yes, exactly the extraneous files ;)